src/app/common/services/messages/messages.service.ts

```diff
@@ -36,9 +36,12 @@ export class MessagesService {
     url = filters.publisher ? url += `&publisher=${filters.publisher}` : url;
     url = filters.subtopic ? url += `&subtopic=${encodeURIComponent(filters.subtopic)}` : url;
     url = filters.name ? url += `&name=${filters.name}` : url;
-    url = filters.value ? url += `&v=${filters.value}` : url;
     url = filters.from ? url += `&from=${filters.from}` : url;
     url = filters.to ? url += `&to=${filters.to}` : url;
+    url = filters.value ? url += `&v=${filters.value}` : url;
+    url = filters.boolValue ? url += `&vb=${filters.boolValue}` : url;
+    url = filters.dataValue ? url += `&vd=${filters.dataValue}` : url;
+    url = filters.stringValue ? url += `&vs=${filters.stringValue}` : url;
 
```

Comment:
Since you have a javascript object and you can iterate through keys, this can be done in a more elegant and general manner. Also, this is error prone, also, because query params can change in the backend.

Reply:
Not sure about this one since not all query parameters follow the name of the filter (for example `v`, `vb`, `vd` and `vs`that after my last commit are `float`, `bool`, `data`, `string`).
In any case if we decide to do it it should be done for all services in a separate PR.

Reply:
`since not all query parameters follow the name of the filter` that's why i suggest to make a mapping between query params and filter names. In js, you use Object to make a map.